src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -117,7 +117,7 @@ private static function prepareRequest(?string $method, ?string $url, array $opt
 
         // Finalize normalization of options
         $options['headers'] = $headers;
-        $options['http_version'] = (string) ($options['http_version'] ?? '');
+        $options['http_version'] = (string) ($options['http_version'] ?? '') ?: null;
```

Comment:
This looks weird. Consider: 

```suggestion
        $options['http_version'] = ((string) $options['http_version']) ?: null;
```

Reply:
not the same: the current code resists http_version being unset